monitor/api/src/test/java/monitorTest/MonitorTest.java

```diff
@@ -0,0 +1,321 @@
+/**
+ * Copyright 2014 SeaClouds
+ * Contact: Dionysis Athanasopoulos <dionysiscsuoi@gmail.com>
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package monitorTest;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.channels.FileChannel;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import resources.Monitor;
+import core.OperatingSystem;
+import core.TxtFileReader;
+import core.TxtFileWriter;
+import core.WindowsBatchFileExecution;
+
+public class MonitorTest {
+
+	private static final String INITIALIZATION_CONFIGURATION_FILE = "./resources/initializationForTesting.properties";
+	private static final String MONITORING_RULES_FILE = "/api/resources/chat-WebApplication-monitoringRules";
+	private static final String DEPLOYMENT_MODEL_FILE = "/api/resources/chat-WebApplication.json";
+	private static final String DATA_COLLECTORS_FILE_NAME = "data-collector-1.3-SNAPSHOT.jar";
+	private static final String DATA_COLLECTORS_INSTALLATION_FILE_NAME = "dataCollectorsInstallation.bat";
+
+	static Logger log = LoggerFactory.getLogger(MonitorTest.class);
+
+	@Test
+	public void test() {
+
+		if (!OperatingSystem.isWindows())
```

Comment:
This test will finish successfully in any other OS different than windows (this is why the tests passes in GH automatically). @dionysiosAthanasopoulos,  does the monitor only work on Windows? If the test refers to a single platform maybe it will be less confusing to name the tests MonitorWindowsTest or something like this. If your software is multiplatform please provide some tests for unix/osx (I think that the incompatibility is that you're using windows batch files to initialize ModaClouds, it should be possible to do the same task by using bash scripting)

Anyway as we need to integrate as soon as possible this PR with the master branch, I will +1 it if you think it's necessary, giving you more time to fix those problems for the next release of the Monitor. 

Thank you very much.